packages/swing-store/test/test-compression.js

```diff
@@ -0,0 +1,398 @@
+    { vatID, startPos: 2, endPos: 5, incarnation: 1 },
+  ]);
+
+  ts.addItem(vatID, 'load-snapshot'); // 5
+  ts.addItem(vatID, 'delivery2'); // 6
+  ts.addItem(vatID, 'save-snapshot'); // 7
+  spanItems.push('load-snapshot\ndelivery2\nsave-snapshot\n');
+  await ss.hostStorage.commit();
+  spans = [...sqlGetSpans.iterate()];
+  items = [...sqlGetItems.iterate()];
+  compressedSpans = [...sqlGetCompressedSpans.iterate()];
+  t.deepEqual(spans, [
+    { vatID, startPos: 0, endPos: 2, incarnation: 0, isCurrent: null },
+    { vatID, startPos: 2, endPos: 5, incarnation: 1, isCurrent: null },
+    { vatID, startPos: 5, endPos: 8, incarnation: 1, isCurrent: 1 },
+  ]);
+  t.deepEqual(items, [
+    { vatID, position: 5, item: 'load-snapshot', incarnation: 1 },
+    { vatID, position: 6, item: 'delivery2', incarnation: 1 },
+    { vatID, position: 7, item: 'save-snapshot', incarnation: 1 },
+  ]);
+  t.deepEqual(compressedSpans, [
+    { vatID, startPos: 0, endPos: 2, incarnation: 0 },
+    { vatID, startPos: 2, endPos: 5, incarnation: 1 },
+  ]);
+
+  ts.rolloverSpan(vatID); // range= 5..8
+  await ss.hostStorage.commit();
+  spans = [...sqlGetSpans.iterate()];
+  items = [...sqlGetItems.iterate()];
+  compressedSpans = [...sqlGetCompressedSpans.iterate()];
+  t.deepEqual(spans, [
+    { vatID, startPos: 0, endPos: 2, incarnation: 0, isCurrent: null },
+    { vatID, startPos: 2, endPos: 5, incarnation: 1, isCurrent: null },
+    { vatID, startPos: 5, endPos: 8, incarnation: 1, isCurrent: null },
+    { vatID, startPos: 8, endPos: 8, incarnation: 1, isCurrent: 1 },
+  ]);
+  t.deepEqual(items, []);
+  t.deepEqual(compressedSpans, [
+    { vatID, startPos: 0, endPos: 2, incarnation: 0 },
+    { vatID, startPos: 2, endPos: 5, incarnation: 1 },
+    { vatID, startPos: 5, endPos: 8, incarnation: 1 },
+  ]);
+
+  ts.addItem(vatID, 'load-snapshot'); // 8
+  ts.addItem(vatID, 'delivery3'); // 9
+  await ss.hostStorage.commit();
+  spans = [...sqlGetSpans.iterate()];
+  items = [...sqlGetItems.iterate()];
+  compressedSpans = [...sqlGetCompressedSpans.iterate()];
+  t.deepEqual(spans, [
+    { vatID, startPos: 0, endPos: 2, incarnation: 0, isCurrent: null },
+    { vatID, startPos: 2, endPos: 5, incarnation: 1, isCurrent: null },
+    { vatID, startPos: 5, endPos: 8, incarnation: 1, isCurrent: null },
+    { vatID, startPos: 8, endPos: 10, incarnation: 1, isCurrent: 1 },
+  ]);
+  t.deepEqual(items, [
+    { vatID, position: 8, item: 'load-snapshot', incarnation: 1 },
+    { vatID, position: 9, item: 'delivery3', incarnation: 1 },
+  ]);
+  t.deepEqual(compressedSpans, [
+    { vatID, startPos: 0, endPos: 2, incarnation: 0 },
+    { vatID, startPos: 2, endPos: 5, incarnation: 1 },
+    { vatID, startPos: 5, endPos: 8, incarnation: 1 },
+  ]);
+
+  const blobItems = sqlGetCompressedSpanBlobs
+    .all()
+    .map(blob => gunzipSync(blob))
+    .map(buf => buf.toString());
+  t.deepEqual(blobItems, spanItems);
+});
```

Comment:
Please add a decompression test which manually populates two spans (one discrete, one compressed, with say two items each) and ensures that `readSpan` can read both. It can create a swingstore normally, then make a secondary `db` object like you do in `test-compression`, then use `db` to insert two spans, two items, and one compressed span, commit, then use `ss.transcriptStore.readSpan` twice, and read out all four entries.

I don't think our normal tests exercise `readSpan` on non-current spans very directly, so if it were somehow broken and unable to handle one type or the other, we might notice without an explicit test like this.